### PR TITLE
feat: cooldown cache & database

### PR DESCRIPTION
**Pull request by darksaid98**
This adds a basic cache for storing cooldowns and support for multiple database drivers so cooldown data can persist between server restarts.

### CODE REVIEW

**Comment by EepyRose** [comment]
Thanks for this! Is there anything more you want to go over with this or would you consider it ready for merge?

**Comment by darksaid98** [comment]
> Thanks for this! Is there anything more you want to go over with this or would you consider it ready for merge?

I'd prefer it if you continued this PR by making commits on the same branch until this is stable.

The db and cache should work perfectly fine, I've not implemented any usage for them, however.

Note that this is also completely untested so it needs a bit of work.

**Comment by darksaid98** [comment]
Fixed the DB issue. I removed `hsql` as a suggested DB driver as it was buggy for me and made `h2` the default.

Also made the reload command run the correct code. See the **"todo"** in the relevant commit for some more stuff needed for reload functionality to properly work. https://github.com/Alathra/AlathranCommands/pull/2/commits/63be11581538e795f556e573042715fc3a84cd93

**Comment by EepyRose** [comment]
The latest two commits were supposed to be on the main branch, my bad heh

**Review by darksaid98 — COMMENTED** [review]
Still some pending TODO's in the project. I recommend you search through it and complete all the relevant ones before release.

File: src/main/java/io/github/Alathra/AlathranCommands/data/model/TPARequest.java [hunk 1]
```diff
@@ -98,6 +105,7 @@ public boolean isExpired() {
      *
      * @return the boolean
      */
+    // TODO correct canAfford to reflect if the teleported player can afford
```
Comment (darksaid98) [inline src/main/java/io/github/Alathra/AlathranCommands/data/model/TPARequest.java]: Is this done?

File: src/main/java/io/github/Alathra/AlathranCommands/runnable/WildTeleportTask.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                        player.teleport(location);
+                    } else {
+                        player.sendMessage(ColorParser.of(TPCfg.get().getString("Messages.error-wildtp-nosuitablelocation")).build());
+                    }
+                });
+            });
+        }
+    }
+
+    void teleport(Player player) {
```
Comment (darksaid98) [inline src/main/java/io/github/Alathra/AlathranCommands/runnable/WildTeleportTask.java]: Empty method.

**Review by darksaid98 — COMMENTED** [review]
Saw your message on Discord. This just needs final testing on the new test server which will be up in the following days.

File: src/main/java/io/github/Alathra/AlathranCommands/data/model/TPARequest.java (hunk 1 above)
Comment (EepyRose) [reply src/main/java/io/github/Alathra/AlathranCommands/data/model/TPARequest.java]: This is fixed as of latest commit.

File: src/main/java/io/github/Alathra/AlathranCommands/runnable/WildTeleportTask.java (hunk 2 above)
Comment (EepyRose) [reply src/main/java/io/github/Alathra/AlathranCommands/runnable/WildTeleportTask.java]: Fixed.

**Pull request closed without merging**